### PR TITLE
CDS-52964-HTTP-step-capability-check

### PR DESCRIPTION
**Pull request by roshnisarangadharan**
Updated capability check section and did self review

# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/http-step.md [hunk 1]
```diff
@@ -117,27 +117,50 @@ In the following examples, the Id of the HTTP step is `HTTP`.
...
 
-## Header Capability Check
+## Header capability check
 
 When Harness runs an HTTP step and connects to a service, it checks to make sure that an HTTP connection can be established.
 
-Some services require that HTTP headers are included in connections. Without the headers, the HTTP connections fail and simple HTTP verification cannot be performed.
+Some services require HTTP headers to be included in connections. Without the headers, the HTTP connections fail and simple HTTP verification cannot be performed.
 
 Harness performs an HTTP header capability check for any header requirements on the target service.
```
Comment (hingerabhinav) [inline docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/http-step.md:134]: Harness performs a HTTP capability check with the headers included on the target service.

File: docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/http-step.md [hunk 2]
```diff
@@ -117,27 +117,50 @@ In the following examples, the Id of the HTTP step is `HTTP`.
...
+Add the required headers in **Headers** and run the deployment. Adding the headers will prevent the `400` error.
+
+Harness secrets are not used during capability checks. They are used in actual steps only. If you have a step configured with URLs and multiple headers like:
+
+`x-api-key : <+secret.getValue('apikey')>`   
+`content-type : application/json`
+
+During a capability check, the non-secret headers are used as is but the secret headers are masked. Harness makes the HTTP request with the URL and headers as follows:
+
+`x-api-key:<api_key>`
```
Comment (hingerabhinav) [inline docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/http-step.md:147]: `x-api-key:<<<api_key>>>`

File: docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/http-step.md [hunk 3]
```diff
@@ -117,27 +117,50 @@ In the following examples, the Id of the HTTP step is `HTTP`.
...
+
+During a capability check, the non-secret headers are used as is but the secret headers are masked. Harness makes the HTTP request with the URL and headers as follows:
+
+`x-api-key:<api_key>`  
+`content-type:application/json`
+
+This results in a `401 Unauthorized` response due to an incorret api key. However, the capability check will be successful and the task will be assigned to the Harness delegate. 
+
+:::note
+Using `<and>` in HTTP requests might result in bad requests on the server side. In such cases, follow these workarounds.
```
Comment (hingerabhinav) [inline docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/http-step.md:153]: not sure what this is `<and>`

File: docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/http-step.md [hunk 4]
```diff
@@ -117,27 +117,50 @@ In the following examples, the Id of the HTTP step is `HTTP`.
...
+
+`x-api-key : <+secret.getValue('apikey')>`   
+`content-type : application/json`
+
+During a capability check, the non-secret headers are used as is but the secret headers are masked. Harness makes the HTTP request with the URL and headers as follows:
+
+`x-api-key:<api_key>`  
+`content-type:application/json`
+
+This results in a `401 Unauthorized` response due to an incorret api key. However, the capability check will be successful and the task will be assigned to the Harness delegate.
```
Comment (hingerabhinav) [inline docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/http-step.md:150]: incorrect

**Review by hingerabhinav — CHANGES_REQUESTED** [review]

File: docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/http-step.md (hunk 3 above)
Comment (roshnisarangadharan) [reply docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/http-step.md:153]: it was mentioned <<<and>>> in the google doc shared with me, https://docs.google.com/document/d/1VKxj12abBk-ouhjI_sEvS_4QmXLuNkNozP_ENCdACjU/edit#heading=h.sotftye1m2yg

Comment (hingerabhinav) [reply docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/http-step.md]: these are specifically <<< and >>>. Let's add those only.

**Review by hingerabhinav — APPROVED** [review]

File: docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/http-step.md [hunk 5]
```diff
@@ -15,15 +15,15 @@ You can use the HTTP step to run HTTP methods containing URLs, methods, headers,
...
 
-For example, if the step name is HTTP and you want to reference the URL entered in its **URL** setting, you would use:
+For example, if the step name is HTTP and you want to reference the URL entered in its **URL** setting, use:
 
 `<+pipeline.stages.tooltips.spec.execution.steps.HTTP.spec.url>`
 
 ## Timeout
 
-The timeout for this step. You can use:
+The timeout for the step. You can use:
```
Comment (SmritiSatya) [inline docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/http-step.md:27]: Timeout for the step. Use:

File: docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/http-step.md [hunk 6]
```diff
@@ -44,51 +44,51 @@ The URL for the HTTP call.
...
 
-## Request Body
+## Request body
 
 The message body of the HTTP message.
 
 ## Assertion
 
-The assertion is used to validate the incoming response. For example, if you wanted to check the health of an HTTP connection, you could use the assertion `<+httpResponseCode> == 200`.
+The assertion is used to validate the incoming response. For example, if you wanted to check the health of an HTTP connection, use the assertion `<+httpResponseCode> == 200`.
```
Comment (SmritiSatya) [inline docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/http-step.md:54]: Can we remove 'the' at the beginning?

**Review by SmritiSatya — DISMISSED** [review]

**Review by SmritiSatya — APPROVED** [review]

**Pull request merged**